### PR TITLE
Improve drill settings

### PR DESCRIPTION
**Pull request by hackerstanislav**
JIRA: RAIL-4583

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10210/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12184/ : SUCCESS in 13m 54s

File: libs/sdk-ui-dashboard/src/presentation/drill/DrillConfigPanel/DrillToUrl/CustomUrlSection.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ type CloseDropdownCallback = (e: React.MouseEvent) => void;
 interface CustomUrlSectionProps {
     toggleModal: ToggleModalCallback;
     closeDropdown: CloseDropdownCallback;
-    urlDrillTarget: UrlDrillTarget;
+    urlDrillTarget?: UrlDrillTarget | null;
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/presentation/drill/DrillConfigPanel/DrillToUrl/CustomUrlSection.tsx:15]: Pls check if ? and null is necessary

File: libs/sdk-ui-dashboard/src/presentation/drill/types.ts [hunk 2]
```diff
@@ -180,7 +180,7 @@ export interface IDrillToCustomUrlConfig {
     customUrl: string;
 }
 
-export function isDrillToCustomUrlConfig(item: UrlDrillTarget): item is IDrillToCustomUrlConfig {
+export function isDrillToCustomUrlConfig(item?: UrlDrillTarget | null): item is IDrillToCustomUrlConfig {
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/presentation/drill/types.ts:183]: isDrillToCustomUrlConfig(item: unknown): item is IDrillToCustomUrlConfig

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/DrillTargets/DrillTargetUrlItem.tsx [hunk 3]
```diff
@@ -164,17 +155,45 @@ export const DrillTargetUrlItem: React.FunctionComponent<DrillUrlItemProps> = (p
...
+
+    const attributeUrlDisplayForms = useDashboardSelector(selectImplicitDrillsToUrlByWidgetRef(widgetRef));
+
+    const drillTargets = useDashboardSelector(selectDrillTargetsByWidgetRef(widgetRef));
+    const allAttributes = useDashboardSelector(selectAllCatalogAttributesMap);
+
+    const attributes = drillTargets?.availableDrillTargets?.attributes;
+
+    const targetAttributes = attributes?.map((drillTarget) =>
+        allAttributes.get({ identifier: drillTarget.attribute.attributeHeader.formOf.identifier }),
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/DrillTargets/DrillTargetUrlItem.tsx:185]: pls provide just ref into allAttributes map 

allAttributes.get(drillTarget.attribute.attributeHeader.formOf.ref)

**Review by ivanmjartan — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3104

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2939/ : FAILURE in 2m 06s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3245/ : SUCCESS in 29s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2090/ : SUCCESS in 19m 19s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12191/ : SUCCESS in 13m 14s

**Pull request merged**